### PR TITLE
Stop audio playback on deletion

### LINKED ISSUE #4627: Deleting audio while playing does not stop the playing
Play an audio file.
Mark this file while playing.
Delete it
Still plays until leaving the conversation.

### PR DESCRIPTION
**Pull request by FeuRenard**
### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Sony Xperia U, Android 4.4.4 (CyanogenMod)
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description
Audio playback will stop if the playing audio attachment gets deleted.

Fixes #4627
// FREEBIE

### CODE REVIEW

**Comment by FeuRenard** [comment]
I changed the code slightly so that AudioSlidePlayer stops the audio directly instead of telling that it is playing currently and then wants to be called again to stop it.

File: src/org/thoughtcrime/securesms/audio/AudioSlidePlayer.java [hunk 1]
```diff
@@ -182,6 +182,15 @@ public synchronized static void stopAll() {
     }
   }
 
+  public static boolean stopIfIsPlaying(@Nullable AudioSlide slide) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/audio/AudioSlidePlayer.java]: think this has to be synchronized

**Comment by FeuRenard** [comment]
There is a reproducible crash with this PR, with and without `synchronized`. I'll have a look at it within the next days and reopen when it is safe.

**Comment by FeuRenard** [comment]
I synchronized the method you requested.
Furthermore I prevent a crash that results from a race condition (see detailed commit message). I'm not very confident that I resolved this race condition in the best way...

**Comment by FeuRenard** [comment]
I'm not very confident with this solution.

**Pull request closed without merging**